text2sql-backend/main.py

```diff
@@ -117,11 +117,9 @@ async def healthcheck() -> SuccessResponse[None] | ErrorResponse:
     return SuccessResponse(status=StatusType.ok)
 
 
-@app.post("/connect", response_model_exclude_none=True)
-async def connect_db(req: ConnectRequest) -> SuccessResponse[dict[str, str]] | ErrorResponse:
-    # Try to connect to provided dsn
+def create_db_connection(dsn: str, name: str) -> SuccessResponse[dict[str, str]] | ErrorResponse:
```

Comment:
I prefer having input models for everything rather than standalone query params like this. Makes thins easier to manage and cleaner and more extendable, and gives you benefits like easily generating typescript models out of your schema models.